src/llama_stack/providers/inline/agents/meta_reference/responses/streaming.py

```diff
@@ -1159,6 +1186,149 @@ async def _process_tools(
                 ):
                     yield stream_event
 
+    async def _process_tool_choice(self) -> str | dict[str, Any] | None:
+        """Process and validate the OpenAI Response tool choice and return the appropriate chat completion tool choice string or object."""
+        if self.ctx.responses_tool_choice:
+            if not self.ctx.chat_tools or len(self.ctx.chat_tools) == 0:
+                logger.warning("No tools available to enforce tool choice, skipping tool choice enforcement")
+                return None
+
+            # retrieve all function tool names from the chat tools
+            # Note: chat_tools contains dicts, not objects
+            chat_tool_names = [
+                tool["function"]["name"] for tool in self.ctx.chat_tools if tool.get("type") == "function"
+            ]
+
+            if isinstance(self.ctx.responses_tool_choice, OpenAIResponseInputToolChoiceMode):
+                if self.ctx.responses_tool_choice.value == "required":
+                    if len(chat_tool_names) == 0:
+                        logger.warning("No tools available to enforce tool choice, skipping tool choice enforcement")
+                        return None
+
+                    # add all function tools to the allowed tools list and set mode to required
+                    return {
```

Comment:
why are we returning untyped dicts? does this correspond to a type somewhere -- feels like it must?

Reply:
I was doing this because the chat completions interface only defines tool_choice as a str or dict so it seemed to make sense to just return a dict here 
see https://github.com/llamastack/llama-stack/blob/main/src/llama_stack_api/inference.py#L965

that being said i've added new named types for these now
let me know if you would prefer to make the chatcompletions interface more strongly typed than it is now as well